### ISSUE TITLE
broken doctests

### ISSUE THREAD

**Issue by grlee77** [PyWavelets/pywt#209]
There are a few things in the recently merged PRs that have broken some existing doctests.  (These can be run by `make doctest`  from within the `doc` folder.  These tests need to either be updated or removed.  Some are trivial to fix, but a few concern me:

One that fails:
```python
from math import sqrt
my_filter_bank = ([sqrt(2)/2, sqrt(2)/2], [-sqrt(2)/2, sqrt(2)/2],
                      [sqrt(2)/2, sqrt(2)/2], [sqrt(2)/2, -sqrt(2)/2])
my_wavelet = pywt.Wavelet('My Haar Wavelet', filter_bank=my_filter_bank)
print(my_wavelet)
```
```
Expected:
    Wavelet My Haar Wavelet
      Family name:
      Short name:
      Filters length: 2
      Orthogonal:     True
      Biorthogonal:   True
      Symmetry:       unknown
Got:
    Wavelet My Haar Wavelet
      Family name:    ]
      Short name:     I
      Filters length: 2
      Orthogonal:     True
      Biorthogonal:   True
      Symmetry:       unknown
      DWT:            True
      CWT:            False
```
The new DWT and CWT fields are to be expected, but why are characters displayed for "Family Name", "Short Name"?  I tried running the example at the command line in Ipython and it actually crashed the kernel!

There is also a test comparing expected coefficients for `db3`.  This test is written to compare 14 decimals and it looks like backing off to 11 decimals would make it pass.  Not sure why this would have changed, though?
```
File "regression/wavelet.rst", line ?, in default
Failed example:
    print_array(w.dec_hi)
Expected:
    [-0.33267055295096, 0.80689150931334, -0.45987750211933, -0.13501102001039, 0.08544127388224, 0.03522629188210]
Got:
    [-0.33267055295008, 0.80689150931109, -0.45987750211849, -0.13501102001025, 0.08544127388203, 0.03522629188571]

**Comment by grlee77** [comment]
I forgot to mention that there are 0 doctest failures if I checkout v0.4.0.

**Comment by grlee77** [comment]
pinging @holgern as these are mostly related to the new wavelets added.
(see `doc/source/regression/wavelet.txt`)

@rgommers do you think it is worth updating the explicit tests for the lists of wavelet names, length of wavelist, etc or just remove those tests?

**Comment by rgommers** [comment]
> @rgommers do you think it is worth updating the explicit tests for the lists of wavelet names, length of wavelist, etc or just remove those tests? 

I think those are pretty pointless as tests. Also as documentation they're not particularly instructive. So I'm fine with removing.

**Comment by grlee77** [comment]
Direct tests of wavelet filter coefficients against Matlab were added by commit a665fe8.  However, accuracy is confirmed at tolerance 5e-11 whereas the doctest for `db3` coefficients compares 14 digits.  As the unit tests have been added we can remove this doctest , but should check if there is a simple reason why the precision changed. 

I can confirm no test failures as of March 7 at commit f71d9d4, but find the coefficient mismatch failures as of Mar10 in 8ff7cfc.

**Comment by grlee77** [comment]
Okay, the coefficient difference was introduced during the C code refactoring when we went from explicitly storing all sets (rec_lo, rec_hi, dec_lo, dec_hi) of filter coefficients, to only storing one and calculating the rest from that.

**Comment by grlee77** [comment]
just a heads up that I plan to make a PR with doctest fixes later today

**Comment by grlee77** [comment]
see #214 for most of the fixes.  still haven't looked into why I get a segfault when calling `print` on a Wavelet object.

**Comment by grlee77** [comment]
The issue with printing the custom wavelet in the first comment is resolved by #218
